addons/l10n_ro_edi/models/ciusro_document.py

```diff
@@ -0,0 +1,198 @@
+    _description = "Document object for tracking CIUS-RO XML sent to E-Factura"
+    _order = 'datetime DESC, id DESC'
+
+    move_id = fields.Many2one('account.move', required=True)
+    state = fields.Selection(
+        selection=[
+            ('to_send', 'To send'),
+            ('sending', 'Sending'),
+            ('sent', 'Sent'),
+            ('nok', 'Not OK'),
+            ('ok', 'OK'),
+            ('error', 'Error'),
+        ],
+        string='E-Factura Status',
+        default='to_send',
+    )
+    attachment_id = fields.Binary(string='E-Factura Attachment')
+    datetime = fields.Datetime()
+    message = fields.Char()
+    key_loading = fields.Char()
+    key_download = fields.Char()
+    key_signature = fields.Char()
+    key_certificate = fields.Char()
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        documents = super().create(vals_list)
+        for document in documents:
+            document.move_id.l10n_ro_edi_document_ids |= document
+            document.move_id.l10n_ro_edi_active_document_id = document
+        return documents
+
+    def _get_attachment_file_name(self):
+        self.ensure_one()
+        return f"{self.move_id.name.replace('/', '_')}_{self.id}.zip"
+
+    def _handle_error(self, message: str):
+        self.state = 'error'
+        self.message = message
+
+    def _make_request(self, endpoint, xml_data=None, session=None):
+        """ Make an API request to E-Factura and return the response
+        :param endpoint: 'upload' (for sending, requires xml_data) |
+                         'stareMesaj' (for fetching status) |
+                         'descarcare' (for downloading answer)
+        :return: response from E-Factura
+        """
+        self.ensure_one()
+        self.datetime = fields.Datetime.now()
+        self.move_id.l10n_ro_edi_active_document_id = self
+
+        url = f"https://api.anaf.ro/test/FCTEL/rest/{endpoint}"
+        headers = {'Content-Type': 'application/xml',
+                   'Authorization': f'Bearer {self.move_id.company_id.l10n_ro_edi_access_token}'}
+
+        match endpoint:
+            case 'upload':
+                if not xml_data:
+                    return self._handle_error(_('CIUS-RO XML not found'))
+                method = 'POST'
+                params = {'standard': 'UBL' if self.move_id.move_type == 'out_invoice' else 'CN',
+                          'cif': self.move_id.company_id.vat.replace('RO', '')}
+                data = xml_data
+            case 'stareMesaj':
+                method = 'GET'
+                params = {'id_incarcare': self.key_loading}
+                data = None
+            case 'descarcare':
+                method = 'GET'
+                params = {'id': self.key_download}
+                data = None
+            case _:
+                raise UserError(_('Invalid request endpoint "%s"') % endpoint)
+
+        try:
+            requester = session if session else requests
```

Comment:
in term of security, it's a very bad practice to have a method getting a complex object as parameter (session) and returning a complex object (response). I suggest another design here (like the one in mexico):
- The invoice is responsible of calling a method on the company telling if you have everything you need to call the web-service (credentials etc). Otherwise, this method should return the errors and the invoice will create a "failed" document.
- The invoice will call the method on the document corresponding to the service needed (upload, stareMesaj... and others web-services names I don't understand because it's not in english...). Then, you call your `_make_requests` api (that should be probably be renamed to something less generic) and return a dictionary with various values like errors or anything you need.
- Back on the invoice, depending if the returned dictionary contains errors or not, created `invoice_sent` or `invoice_sending_failed` document